app/tests/evaluation_tests/test_models.py

```diff
@@ -740,3 +742,110 @@ def test_give_algorithm_editors_job_view_permissions_only_for_algorithm_phase():
 
     phase.submission_kind = Phase.SubmissionKindChoices.ALGORITHM
     phase.full_clean()
+
+
+@pytest.mark.django_db
+def test_parent_phase_choices():
+    p1, p2, p3, p4, p5 = PhaseFactory.create_batch(
+        5, challenge=ChallengeFactory()
+    )
+    p6 = PhaseFactory()
+
+    for phase in [p1, p2, p3, p4, p6]:
+        phase.submission_kind = SubmissionKindChoices.ALGORITHM
+    p5.submission_kind = SubmissionKindChoices.CSV
+
+    ci1, ci2, ci3, ci4 = ComponentInterfaceFactory.create_batch(4)
+
+    for phase in [p1, p2, p3, p4, p5, p6]:
+        phase.algorithm_inputs.set([ci1])
+
+    for phase in [p1, p4, p5]:
+        phase.algorithm_outputs.set([ci2, ci3])
+    p2.algorithm_outputs.set([ci2, ci4])
+    p3.algorithm_outputs.set([ci2])
+
+    for phase in [p1, p2, p3, p4, p5, p6]:
+        phase.save()
+
+    assert list(p1.parent_phase_choices) == [p4]
+
+
+@pytest.mark.django_db
+def test_parent_phase_choices_no_circular_dependency():
+    p1, p2, p3, p4 = PhaseFactory.create_batch(
+        4,
+        challenge=ChallengeFactory(),
+        submission_kind=SubmissionKindChoices.ALGORITHM,
+    )
+    ci1, ci2 = ComponentInterfaceFactory.create_batch(2)
+
+    for phase in [p1, p2, p3, p4]:
+        phase.algorithm_inputs.set([ci1])
+        phase.algorithm_outputs.set([ci2])
+
+    p1.parent = p2
+    p2.parent = p3
+    p3.parent = p4
+    p1.save()
+    p2.save()
+    p3.save()
+
+    for p in [p2, p3, p4]:
+        assert p in p1.parent_phase_choices
+    assert p1 not in p1.parent_phase_choices
+    for p in [p3, p4]:
+        assert p in p2.parent_phase_choices
+    assert p1 not in p2.parent_phase_choices
```

Comment:
Wouldn't doing a set comparison be clearer?

```suggestion
    assert set(p1. parent_phase_choices) == {p2, p3, p4}
    assert set(p2.parent_phase_choices) == {p3, p4}
```

Reply:
Ah I was looking for a better way but doing a set comparison didn't occur to me. Thanks!